collector/mdadm_linux.go

```diff
@@ -99,17 +100,30 @@ var (
 		[]string{"device"},
 		nil,
 	)
+
+	mdraidDisks = prometheus.NewDesc(
+		prometheus.BuildFQName(namespace, "md", "raid_disks"),
+		"Number of raid disks on device.",
```

Comment:
Maybe we should have a label 'degraded' instead of a separate metric? Don't feel strong either way though

Reply:
IIUC That might convey degradation for the `raid_disks` in a manner rather than the `device`. I believe a separate metric works here since the logic isn't tied between labels but only for the metric, i.e., `metricFoo{labelA="x", labelAA="y"}` where `labelAA` solely provides information about the other label rather than the metric v/s `metricBar{labelA="x"}` and `metricBaz{labelAA="y"}` (the current approach).

LMK if you think otherwise. :)

Reply:
Ok makes sense